pkg/api/customization/catalog/template.go

```diff
@@ -3,20 +3,68 @@ package catalog
 import (
 	"bytes"
 	"encoding/base64"
+	"fmt"
 	"net/http"
 	"strings"
 	"time"
 
+	"github.com/rancher/norman/types/convert"
+
 	"github.com/rancher/norman/api/access"
 	"github.com/rancher/norman/httperror"
 	"github.com/rancher/norman/types"
 	"github.com/rancher/rancher/pkg/templatecontent"
 	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
 	managementschema "github.com/rancher/types/apis/management.cattle.io/v3/schema"
 	client "github.com/rancher/types/client/management/v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func TemplateFormatter(apiContext *types.APIContext, resource *types.RawResource) {
+func checkUpgrade(t TemplateWrapper, templateVersionName string) (string, error) {
```

Comment:
checkUpgrade is a bad name with very little context in the future. Any idea for a better name?

Reply:
How about `getChartIconURL` ?